spyder/plugins/updatemanager/workers.py

```diff
@@ -445,7 +480,144 @@ def start(self):
+
+        self.asset_info = _check_asset_available(releases, updater=True)
+
+    def _clean_installer_dir(self):
+        """Remove downloaded file"""
+        installer_dir = osp.dirname(self.installer_path)
+        if osp.exists(installer_dir):
+            try:
+                shutil.rmtree(installer_dir)
+            except OSError as err:
+                logger.debug(err, stack_info=True)
+
+    def _download_asset(self):
+        """Download Updater lock file"""
+        self.installer_path = osp.join(
+            get_temp_dir(), "updates", "spyder-updater",
+            str(self.asset_info["version"]), self.asset_info["filename"]
+        )
+
+        if (
+            osp.exists(self.installer_path)
+            and validate_download(
+                self.installer_path, self.asset_info["checksum"]
+            )
+        ):
+            logger.debug(f"{self.installer_path} already downloaded.")
+            return
+
+        self._clean_installer_dir()
+        dirname = osp.dirname(self.installer_path)
+        os.makedirs(dirname, exist_ok=True)
+
+        url = self.asset_info["url"]
+        logger.info(f"Downloading {url} to {self.installer_path}")
+
+        page = requests.get(url, headers=GH_HEADERS)
+        page.raise_for_status()
+        with open(self.installer_path, 'wb') as f:
+            f.write(page.content)
+
+        if validate_download(self.installer_path, self.asset_info["checksum"]):
+            logger.info('Download successfully completed.')
+            with ZipFile(self.installer_path, 'r') as f:
+                f.extractall(dirname)
+        else:
+            raise UpdateDownloadError("Download failed!")
+
+    def _install_update(self):
+        """Install or update spyder-updater environment"""
+        dirname = osp.dirname(self.installer_path)
+
+        if os.name == "nt":
+            plat = "win-64"
+        if sys.platform == "darwin":
+            plat = "osx-arm64" if platform.machine() == "arm64" else "osx-64"
+        else:
+            plat = "linux-64"
+        spy_updater_lock = osp.join(dirname, f"conda-updater-{plat}.lock")
+        spy_updater_conda = osp.join(dirname, "spyder-updater*.conda")
+
+        conda_exe = find_conda()
+        conda_cmd = "update" if UPDATER_VERSION > parse("0.0.0") else "create"
+        env_path = osp.join(osp.dirname(sys.prefix), "spyder-updater")
+        cmd = [
+            # Update spyder-updater environment
+            conda_exe,
+            conda_cmd, "--yes",
+            "--prefix", env_path,
+            "--file", spy_updater_lock,
+            "&&",
+            # Update spyder-updater
+            conda_exe,
+            "install", "--yes",
+            "--prefix", env_path,
+            "--no-deps",
+            "--force-reinstall",
+            spy_updater_conda
+        ]
+        logger.debug(f"""Conda command: '{" ".join(cmd)}'""")
+        proc = run(" ".join(cmd), shell=True, capture_output=True, text=True)
```

Comment:
```suggestion
        proc = subprocess.run(
            " ".join(cmd), shell=True, capture_output=True, text=True
        )
```

To go with my suggestion above of using `subprocess.run` instead of `run`.